src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -3196,7 +3196,12 @@ public function set_attribute( $name, $value ): bool {
 			 *
 			 * @see https://html.spec.whatwg.org/#attributes-3
 			 */
-			$escaped_new_value = in_array( $comparable_name, wp_kses_uri_attributes() ) ? esc_url( $value ) : esc_attr( $value );
+			if ( 'src' === $comparable_name && strpos( $value, 'data:' ) === 0 ) {
```

Comment:
```suggestion
			if ( 'src' === $comparable_name && str_starts_with( $value, 'data:' ) ) {
```

Reply:
Thanks for the suggestion. I have addressed it.